seml/queuing.py

```diff
@@ -138,11 +138,10 @@ def queue_experiments(db_collection_name, config_file, force_duplicates, no_hash
     else:
         batch_id = batch_id + 1
 
-    if "project_root_dir" not in seml_config:
-        logging.warning("'project_root_dir' not defined in seml config. Source files will not be uploaded.")
-        uploaded_files = None
-    else:
+    if seml_config['has_root_dir']:
```

Comment:
This is only used here? Why not call it something like `saved_sources`?

Reply:
Or can we just check for `seml['source_files']` instead?

Reply:
> Or can we just check for seml['source_files'] instead?
  
Unfortunately not, since this is where we would upload the sources.

We can call it `sources_saved`, but then I guess it would make sense to have the option to disable uploading from the command line. What do you think?

Reply:
Yes, so `seml['source_files']` is present exactly when we have saved source files. So we only need to check `if 'source_files' in seml_config`, right?

Reply:
This won't work here, because this is before we've uploaded the files and created the entry `seml['source_files']`. Or am I missing something?

Reply:
Oh, yes, that's true. So it's just about a better name then. How about `use_uploaded_sources`?